coalahtml/_coalahtml/app/controllers/files.js

```diff
@@ -33,7 +33,31 @@ angular.module('coalaHtmlApp')
     };
     $scope.fileName = $routeParams.fileName || $rootScope.ROOTHOME;
     parseCoalaProject($scope.fileName);
-    $scope.basename = function(path_name){
+
+    $scope.basename = function(path_name) {
+      path_name = path_name || $rootScope.ROOTHOME;
       return path_name.split('/').slice(-1)[0];
     };
+    var len = function(path_name) {
+      path_name = path_name || $rootScope.ROOTHOME;
+      return path_name.split('/').length;
+    };
```

Comment:
I am calculating the same thing (https://github.com/coala-analyzer/coala-html/pull/52/commits/4922d4c32e9d92ba2d9764ef3f6a3cdb035d93b9#diff-b803c22fe9d790db802829435dd58480R53)[2 times]. To make the `getNameLink` clean, I need to separate the process. 
I agree, the name is ambiguous. I will change the name.